### DISCUSSION TITLE
Possible to have pseudo-permalink URL to download artifacts?

### DISCUSSION THREAD

**Discussion by jonlamb-gh** [keygen-sh/community#26] [General]
I'm revisiting some of our licensing, and upgrading from the now deprecated Dist API along the way.

We were previously using
```
https://dist.keygen.sh/v1/{ACCOUNT}/{PRODUCT}/releases/{PLATFORM}/{RELEASE}/?key={LICENSE_KEY_ONLY_FOR_DOWNLOADS}
```
to provide a pseudo-permalink on our website for downloading (`LICENSE_KEY_ONLY_FOR_DOWNLOADS` is just a dummy license with no entitlements to enable the download).

Since our product does all the real licensing logic within, we don't need any gating around the downloading part.

If I'm reading the new [download release artifact docs](https://keygen.sh/docs/api/releases/#releases-artifacts-download) correctly, it seems like there's no way to set a NULL/infinite TTL and that it's geared more towards a temporary link.

Is it possible to do provide such a thing using the newer APIs?

**Comment by ezekg** [comment]
You can set your product's [distribution strategy](https://keygen.sh/docs/api/products/#products-object-attrs-distributionStrategy) to `OPEN`, which will allow you to distribute releases without authentication, i.e. without a license key. This is great when you're handling licensing in-app like in your case.

Then you can use a URL like the following, instead of using a dummy license:

```
https://get.keygen.sh/{ACCOUNT}/{ARTIFACT}
```

This will generate and redirect to a temporary download link each time the URL is accessed. (I.e. there's no need for a longer `ttl`, since it's generated on the fly per-request, each time the URL is accessed.)

For example, this is how we distribute [Keygen's CLI](https://keygen.sh/docs/cli/):

```
curl -sSL https://get.keygen.sh/keygen/cli/install.sh | sh
```

**Reply by jonlamb-gh** [reply]
Ah, perfect.
Thanks!

**Reply by JonHollander** [reply]
Is it possible to use this URL format to access the latest release from a beta channel?  I thought it would be something like `https://get.keygen.sh/{ACCOUNT_OR_SLUG}/{PRODUCT}/{CHANNEL}/{ARTIFACT_NAME}`, but that doesn't seem to work.

**Reply by ezekg** [reply]
Use [the `?channel=beta` query parameter](https://keygen.sh/docs/api/artifacts/#artifacts-retrieve-query-channel).
